packages/slice-machine/src/features/onboarding/OnboardingProvider.tsx

```diff
@@ -0,0 +1,85 @@
+import { createContext, ReactNode, useContext } from "react";
+
+import { onboardingSteps as steps } from "@/features/onboarding/content";
+import type {
+  OnboardingStep,
+  OnboardingStepStatuses,
+  OnboardingStepType,
+} from "@/features/onboarding/types";
+import { usePersistedState } from "@/hooks/usePersistedState";
+
+type OnboardingContext = {
+  steps: OnboardingStep[];
+  completedStepCount: number;
+  toggleStepComplete: (step: OnboardingStepType) => void;
+  getStepIndex: (stepOrId: OnboardingStep | OnboardingStepType) => number;
+  isStepComplete: (stepOrId: OnboardingStep | OnboardingStepType) => boolean;
+  isComplete: boolean;
+};
+
+export const OnboardingContext = createContext<OnboardingContext | undefined>(
+  undefined,
+);
+
+const getInitialState = (): OnboardingStepStatuses => {
+  // if the old guide was dismissed, all steps start as complete
+  const startComplete =
```

Comment:
💡 #idea: tiny thing, but maybe this could be renamed to a `isOnboardingCompleted`, or `isCompleted`

Reply:
I prefer to mention that it's regarding the previous guide, but I renamed it to be more clear (`wasOldGuideDismissed`)

Reply:
Regarding the old guide, we discussed not displaying the new one if the user had previously dismissed it before, that's why I'm starting with all completed. This value is also only used if there isn't already something stored. Not sure if that's the issue you're describing 😄

Reply:
I was thinking about keeping one more piece of persistent state -> that whole onboarding was completed, then we can exit early, without computing each step etc.
On the other hand there is not much things to process anyway so the way you did is perfectly fine I guess.

Reply:
I don't think it's necessary, it's not too much to compute and having an extra value means having multiple sources of truth.